sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -97,8 +97,18 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    */
   @throws[AnalysisException]("database does not exist")
   override def listTables(dbName: String): Dataset[Table] = {
-    val tables = sessionCatalog.listTables(dbName).map(makeTable)
-    CatalogImpl.makeDataset(tables, sparkSession)
+    if (sessionCatalog.databaseExists(dbName)) {
+      val tables = sessionCatalog.listTables(dbName).map(makeTable)
+      CatalogImpl.makeDataset(tables, sparkSession)
+    } else {
+      val multiParts = sparkSession.sessionState.sqlParser.parseMultipartIdentifier(dbName)
+      val plan = ShowTables(UnresolvedNamespace(multiParts), None)
+      val ret = sparkSession.sessionState.executePlan(plan).toRdd.collect()
+      val tables = ret
+        .map(row => TableIdentifier(row.getString(1), Some(row.getString(0))))
+        .map(makeTable)
```

Comment:
`makeTable` only looks up table from hive metastore. I think we need a new `makeTable` which takes `Seq[String]` and looks up table through analyzer.

Reply:
oh yes you are right. Let me add a version that looks up through analyzer.

Reply:
done